cpp/src/arrow/compute/exec/aggregate_node.cc

```diff
@@ -570,8 +773,12 @@ class GroupByNode : public ExecNode, public TracedNode {
     return output_->InputReceived(this, out_data_.Slice(batch_size * n, batch_size));
   }
 
-  Status OutputResult() {
-    auto scope = TraceFinish();
+  Status OutputResult(bool is_last = false, bool traced = false) {
```

Comment:
What does is_last and traced do here?

Reply:
`is_last` is `true` when this is the last time `OutputResult` is called in this node. In this case, when `traced` is also `true`, the invocation is traced using the facilities inherited from `TracedNode`.

Reply:
Removed this piece of the code.